### PR TITLE
CANVAS-114 - Admin panel: Make terms a dropdown

### PR DESCRIPTION
**Pull request by chern**

### CODE REVIEW

File: src/client/components/AdminPanel/AdminListings.js [hunk 1]
```diff
@@ -105,89 +122,61 @@ export const AdminListings = ({ mediaType, setError }) => {
...
-              <SimpleSelect
-                renderLabel="Subject Area"
-                value={selectedSubjectArea}
-                onChange={handleSubjectSelect}
-                visibleOptionsCount={5}
-                width="12rem"
-              >
-                <SimpleSelect.Option id="all" value="">
-                  All
+      <Flex margin="none none none">
```
Comment (maxpluez) [inline src/client/components/AdminPanel/AdminListings.js:129]: Just curious, does margin="none none none" make any difference?

File: src/client/components/AdminPanel/AdminListingsToggle.js [hunk 2]
```diff
@@ -34,9 +41,9 @@ export const AdminListingsToggle = ({ shortname, listings, mediaType }) => {
   // Summary includes the shortname and the number of listings
   // If missing shortname, include an orange warning pill
   const shortnameSummary = (
-    <>
+    <View>
```
Comment (maxpluez) [inline src/client/components/AdminPanel/AdminListingsToggle.js:44]: I figured <> </> is better than wrapping with a < View >. I'll create a ticket to get this change done (change all <View> without props to <>)

File: src/server/services/utility.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+module.exports.sortAcademicTerms = function(termsArray, descending = false) {
+  const termsEnumArray = [];
+  const termsEnumMap = new Map();
+  for (const term of termsArray) {
+    const enumeratedTerm = enumerateAcademicTerm(term);
+    termsEnumArray.push(enumeratedTerm);
+    termsEnumMap.set(enumeratedTerm, term);
+  }
+
+  termsEnumArray.sort();
```
Comment (maxpluez) [inline src/server/services/utility.js:47]: You can directly sort termsArray by using a callback function in sort, something like const result = termsArray.sort(callback). Would it make the code here cleaner?

File: src/client/components/AdminPanel/AdminListings.js (hunk 1 above)
Comment (chern) [reply src/client/components/AdminPanel/AdminListings.js:129]: Oh yeah, good catch, forgot to remove it when I was tinkering with the margin

File: src/client/components/AdminPanel/AdminListingsToggle.js (hunk 2 above)
Comment (chern) [reply src/client/components/AdminPanel/AdminListingsToggle.js:44]: What's the reason <></> is better than <View>?

Comment (maxpluez) [reply src/client/components/AdminPanel/AdminListingsToggle.js]: So <View> renders a lot of unnecessary html components.
See https://www.barrymichaeldoyle.com/fragment/

File: src/client/components/AdminPanel/AdminListings.js [hunk 4]
```diff
@@ -65,24 +83,23 @@ export const AdminListings = ({ mediaType, setError }) => {
         });
       });
```
[the remark this answers is not in the document]
Comment (maxpluez) [reply src/client/components/AdminPanel/AdminListings.js:84]: Oh github collapsed the catch statement for me so I didn't see it. All good then.

File: src/server/services/VideoresServices.js [hunk 5]
```diff
@@ -52,11 +53,19 @@ class VideoresServices {
...
+    );
     return termMedia;
   }
 
+  static async getTerms() {
+    const terms = await MediaQuery.getTerms('videoreserves');
+    terms.sort(compareAcademicTerms).reverse();
+    return terms;
+  }
+
```
Comment (maxpluez) [inline src/server/services/VideoresServices.js:68]: Seems like Bruincast services, video res services, and music res services all use the same functions. Can we put it inside media resource services and reuse for all 3 cases?

File: src/server/api/medias/videores.js [hunk 6]
```diff
@@ -34,4 +33,11 @@ router.get('/subjectareas', (req, res) => {
   );
 });
 
+router.get('/terms', (req, res) => {
+  if (!CheckRoleServices.isAdmin(res.locals.token.roles)) {
+    return res.status(403).send(new Error('Unauthorized role'));
+  }
+  VideoresServices.getTerms().then(terms => res.send(terms));
+});
+
```
Comment (maxpluez) [inline src/server/api/medias/videores.js:42]: Seems like the 3 /terms are doing the same thing. Is it possible to move to /api/medias/index.js route?

**Comment by maxpluez** [comment]
Screenshot from 2020-09-02 11-27-52
Tested. Looks good. Moving to passed.

**Pull request merged**